test/reporters/tests/github-actions.test.ts

```diff
@@ -1,17 +1,41 @@
 import { resolve } from 'pathe'
-import { expect, test } from 'vitest'
+import { describe, expect, it } from 'vitest'
 import { GithubActionsReporter } from '../../../packages/vitest/src/node/reporters'
 import { runVitest } from '../../test-utils'
 
-test(GithubActionsReporter, async () => {
-  let { stdout, stderr } = await runVitest(
-    { reporters: new GithubActionsReporter(), root: './fixtures', include: ['**/some-failing.test.ts'] },
-  )
-  stdout = stdout.replace(resolve(__dirname, '..').replace(/:/g, '%3A'), '__TEST_DIR__')
-  expect(stdout).toMatchInlineSnapshot(`
+describe(GithubActionsReporter, () => {
+  it('uses absolute path by default', async () => {
+    let { stdout, stderr } = await runVitest(
+      { reporters: new GithubActionsReporter(), root: './fixtures', include: ['**/some-failing.test.ts'] },
+    )
+    stdout = stdout.replace(resolve(__dirname, '..').replace(/:/g, '%3A'), '__TEST_DIR__')
+    expect(stdout).toMatchInlineSnapshot(`
     "
     ::error file=__TEST_DIR__/fixtures/some-failing.test.ts,title=some-failing.test.ts > 3 + 3 = 7,line=8,column=17::AssertionError: expected 6 to be 7 // Object.is equality%0A%0A- Expected%0A+ Received%0A%0A- 7%0A+ 6%0A%0A ❯ some-failing.test.ts:8:17%0A%0A
     "
   `)
-  expect(stderr).toBe('')
+    expect(stderr).toBe('')
+  })
+
+  it('uses onWritePath to format path', async () => {
+    let { stdout, stderr } = await runVitest(
+      {
+        reporters: new GithubActionsReporter({
+          onWritePath(path) {
+            return path.replace(resolve(__dirname, '..').replace(/:/g, '%3A'), '__TEST_DIR__')
+          },
+        }),
+        root: './fixtures',
+        include: ['**/some-failing.test.ts'],
+        env: { GITHUB_WORKSPACE: '/home/user/workspace' },
+      },
+    )
+    stdout = stdout.replace(resolve(__dirname, '..').replace(/:/g, '%3A'), '__TEST_DIR__')
+    expect(stdout).toMatchInlineSnapshot(`
+      "
+      ::error file=__TEST_DIR__/fixtures/some-failing.test.ts,title=some-failing.test.ts > 3 + 3 = 7,line=8,column=17::AssertionError: expected 6 to be 7 // Object.is equality%0A%0A- Expected%0A+ Received%0A%0A- 7%0A+ 6%0A%0A ❯ some-failing.test.ts:8:17%0A%0A
+      "
+    `)
```

Comment:
This assertion is 100% identical with `it('uses absolute path by default'`. We don't really see the usage of `onWritePath` here. Does this test pass locally as it's relying on `process.env.GITHUB_WORKSPACE`?

I think we could instead verify that some non-existing path written in `onWritePath` ends up in the report.